### PR TITLE
Get, Add and Delete Books | API

### PR DESCRIPTION
**Pull request by thasipablo**
- Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as backend.
- Use [`axios`](https://axios-http.com/docs/intro) with [`createAsyncThunk`](https://redux-toolkit.js.org/api/createAsyncThunk) to make HTTP requests to the backend
- Fetch `books` from the Bookstore API **after** the component has rendered
- Refactor **add book** and **remove book** features to persist changes in the server

### CODE REVIEW

File: src/components/BookComponent.jsx
Comment (Efiamotu-1) [inline src/components/BookComponent.jsx:1]: - [ ] In your readme file under the `License` section, when I try to click the link to show your license file, the link is currently broken as it displays a 404 error page, please I kindly suggest you fix this by providing the appropriate link, it is unprofessional to leave broken links in your readme file 👍

File: src/components/BookFormComponent.jsx [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
+          author,
+          category,
+          item_id: itemId,
+        }),
+      );
       setTitle('');
       setAuthor('');
+      setCategory('');
     }
   };
```
Comment (Efiamotu-1) [inline src/components/BookFormComponent.jsx:28]: - [ ] When a new book is added to your list of books, the new book that is added should be displayed on the webpage, however this is not the way your app currently functions as it only adds the book to the API and the newly added book is not displayed unless the page is reloaded, please kindly add this functionality to your project and resubmit for approval 👍 ✨

File: src/components/BookComponent.jsx [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-        Author:
-        {book.author}
+        <span>Author: </span>
+        <span>{book.author}</span>
+      </p>
+      <p>
+        <span>Category: </span>
+        <span>{book.category}</span>
       </p>
       <button type="button" onClick={() => dispatch(removeBook(book.item_id))}>
```
Comment (Efiamotu-1) [inline src/components/BookComponent.jsx:20]: - [ ] When a book is removed from your list of books, the book that is removed should ber on the removed on the webpage, however this is not the way your app currently functions as it only deletes the book when the API request is made  and the newly removed book is not removed unless the page is reloaded, please kindly add this functionality to your project and resubmit for approval 👍 ✨

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @thasipablo , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by thasipablo** [comment]
Thanks @Efiamotu-1 
All improvements have been implemented 
- Add license file 
- Refresh interface / book list when submitting and deleting.

**Review by Gambit142 — APPROVED** [review]
Hi @thasipablo 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Setup linters correctly ✔️ 
- Great use of `createAsyncThunk` method ✔️ 
- Fetched data from the required API ✔️ 
- Books can be deleted successfully ✔️ 
- Books can be added successfully ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**